geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/misc/WANSSLDUnitTest.java

```diff
@@ -14,107 +14,624 @@
+    cache = CacheFactory.create(ds);
+  }
 
-    vm4.invoke(() -> WANTestBase.createCache(lnPort));
+  private void createCacheWithSSL(Integer locPort) {
+    Properties props = getDistributedSystemProperties();
+    props.putAll(getBasicMemberProperties());
+    props.putAll(getSSLProperties());
 
-    String senderId = "ln";
-    vm4.invoke(
-        () -> WANTestBase.createSender(senderId, 2, false, 100, 10, false, false, null, true));
+    props.put(ConfigurationProperties.LOCATORS, "localhost[" + locPort + "]");
 
-    String regionName = getTestMethodName() + "_RR";
-    vm2.invoke(() -> WANTestBase.createReplicatedRegion(regionName, null, isOffHeap()));
+    logger.info("Starting cache ds with following properties \n" + props);
 
-    vm4.invoke(() -> WANTestBase.startSender(senderId));
+    InternalDistributedSystem ds = getSystem(props);
+    cache = CacheFactory.create(ds);
+  }
 
-    // Verify the sender is started
-    vm4.invoke(() -> verifySenderRunningState(senderId));
+  private Properties getSSLProperties() {
+    Properties props = new Properties();
+    props.put(ConfigurationProperties.GATEWAY_SSL_ENABLED, "true");
+    props.put(ConfigurationProperties.GATEWAY_SSL_PROTOCOLS, "any");
+    props.put(ConfigurationProperties.GATEWAY_SSL_CIPHERS, "any");
+    props.put(ConfigurationProperties.GATEWAY_SSL_REQUIRE_AUTHENTICATION, "true");
+    props.put(ConfigurationProperties.GATEWAY_SSL_KEYSTORE_TYPE, "jks");
+    props.put(ConfigurationProperties.GATEWAY_SSL_KEYSTORE,
+        TestUtil.getResourcePath(WANSSLDUnitTest.class,
+            "/org/apache/geode/cache/client/internal/default.keystore"));
+    props.put(ConfigurationProperties.GATEWAY_SSL_KEYSTORE_PASSWORD, "password");
+    props.put(ConfigurationProperties.GATEWAY_SSL_TRUSTSTORE,
+        TestUtil.getResourcePath(WANSSLDUnitTest.class,
+            "/org/apache/geode/cache/client/internal/default.keystore"));
+    props.put(ConfigurationProperties.GATEWAY_SSL_TRUSTSTORE_PASSWORD, "password");
+    return props;
+  }
 
-    // Verify the sender is not connected
-    vm4.invoke(() -> verifySenderConnectedState(senderId, false));
+  private int createReceiverWithSSL(int locPort) {
+    Properties props = getDistributedSystemProperties();
+
+    Properties p = getSSLProperties();
+    props.putAll(p);
+    props.putAll(getBasicMemberProperties());
+    props.put(ConfigurationProperties.LOCATORS, "localhost[" + locPort + "]");
+
+    logger.info("Starting cache ds with following properties \n" + props);
+
+    InternalDistributedSystem ds = getSystem(props);
+    cache = CacheFactory.create(ds);
+    GatewayReceiverFactory fact = cache.createGatewayReceiverFactory();
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    fact.setStartPort(port);
+    fact.setEndPort(port);
+    fact.setManualStart(true);
+    GatewayReceiver receiver = fact.create();
+    try {
+      receiver.start();
+    } catch (IOException e) {
```

Comment:
Another unexpected exception to not catch. If this is really problematic, you could do this instead:
```
try {
  ...
} catch (IOException e) {
  throw new UncheckedIOException(e);
}
```